api-reference/10 UI Widgets/dxDataGrid/1 Configuration/onExporting.md

```diff
@@ -27,137 +27,216 @@ The name of the file where grid data is about to be exported.
     $(function() {
-        $("#dataGridContainer").dxDataGrid({
+        $('#gridContainer').dxDataGrid({
             // ...
+            export: {
+                enabled: true
+            },
             columns: [{
-                dataField: "ID",
+                dataField: 'ID',
                 visible: false
-            }, // ...
-            ]
-            onExporting: function (e) {
+            }, {
+                // ...
+            }],
+            onExporting: function(e) { 
                 e.component.beginUpdate();
-                e.component.columnOption("ID", "visible", true);
-            },
-            onExported: function (e) {
-                e.component.columnOption("ID", "visible", false);
-                e.component.endUpdate();
+                e.component.columnOption('ID', 'visible', true);
+
+                var workbook = new ExcelJS.Workbook(); 
+                var worksheet = workbook.addWorksheet('Main sheet'); 
+                DevExpress.excelExporter.exportDataGrid({ 
+                    worksheet: worksheet, 
+                    component: e.component
+                }).then(function() {
+                    workbook.xlsx.writeBuffer().then(function(buffer) { 
+                        saveAs(new Blob([buffer], { type: 'application/octet-stream' }), 'DataGrid.xlsx'); 
+                    }); 
+                }).then(() => {
+                    e.component.columnOption('ID', 'visible', false);
+                    e.component.endUpdate();
+                }); 
+                e.cancel = true; 
             }
         });
     });
 
-##### Angular
+##### Angular   
 
-    <!--TypeScript-->
-    import { DxDataGridModule } from "devextreme-angular";
-    // ...
+    <!-- tab: app.component.html -->
+    <dx-data-grid ...
+        (onExporting)="onExporting($event)">
+        <dxo-export [enabled]="true"></dxo-export>
+        <dxi-column dataField="ID" [visible]="false"></dxi-column>
+    </dx-data-grid>
+
+    <!-- tab: app.component.ts -->
+    import { Component } from '@angular/core';
+    import { exportDataGrid } from 'devextreme/excel_exporter';
+    import ExcelJS from 'exceljs';
+    import saveAs from 'file-saver';
+    
+    @Component({
+        selector: 'app-root',
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
     export class AppComponent {
-        onExporting (e) {
+        onExporting(e) {
             e.component.beginUpdate();
             e.component.columnOption("ID", "visible", true);
-        };
-        onExported (e) {
-            e.component.columnOption("ID", "visible", false);
-            e.component.endUpdate();
+            const workbook = new ExcelJS.Workbook();
+            const worksheet = workbook.addWorksheet("Employees");
+
+            exportDataGrid({
+                component: e.component,
+                worksheet: worksheet
+                }).then(function() {
+                workbook.xlsx
```

Comment:
In this code example need to fix the indentations. 
Check it also in all approaches, pls